### ISSUE TITLE
`fallible-context-manager` (`RUF075`) - false positive when `yield` is inside `for` loop with no cleanup afterwards

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#25378]
### Summary

```py
from contextlib import contextmanager

@contextmanager
def foo(items: list[int]):
    for value in items:
        yield # RUF075: Context manager does not catch exceptions
        break
```

```toml
[lint]
preview = true
select = [ "RUF075" ]
```

[Playground](https://play.ruff.rs/9167c867-6193-496d-b2ef-1075be51083e)

as discussed in https://github.com/astral-sh/ruff/issues/15629#issuecomment-2604900179 we wanted to avoid false positives when `yield` is the last statement, but it would be nice if it was also able to detect if there's a `break` statement right after it when the surrounding `for` loop is the last statement

### Version

0.15.14

**Comment by ntBre** [comment]
Ah yeah, that makes sense! This example currently works with `return` instead of `break`, but hopefully we can handle `break` too without too much trouble.
